test/test_jit.py

```diff
@@ -5013,6 +5013,87 @@ def test(op, const, swap_args):
     def test_tensor_number_math(self):
         self._test_tensor_number_math()
 
+    def test_torch_tensor_bad_input(self):
+        with self.assertRaisesRegex(RuntimeError, "Input list to torch.tensor must be of ints, floats, "
+                                    "or bools, got None"):
+            @torch.jit.script
+            def test():
+                return torch.tensor([None])
+
+        with self.assertRaisesRegex(RuntimeError, "Note: empty lists are constructed as Tensor\[\]"):
+            @torch.jit.script
+            def foo():
+                return torch.tensor([])
+
+        @torch.jit.script
+        def foo():
+            return torch.tensor([[2, 2], [1]])
+        with self.assertRaisesRegex(RuntimeError, "Expected sequence of length"):
+            foo()
+
+    def test_torch_tensor_empty_list(self):
+        code = dedent('''
+        def func():
+            return torch.tensor(torch.jit.annotate(List[int], []))
+        ''')
+        scope = {}
+        exec(code, globals(), scope)
+        cu = torch.jit.CompilationUnit(code)
+        t1 = cu.func()
+        t2 = scope['func']()
+
+        # torchscript returns int tensor, python returns float tensor
+        self.assertNotEqual(t1.dtype, t2.dtype)
```

Comment:
Note: python only sees torch.tensor([]), which has a default type of float.  Script uses the annotated type of the list, which is int.

Reply:
We shouldn't have this discrepancy? Or at least, there should be a warning here

Reply:
The first time a user inputs torch.tensor([]) will result in exception saying they have to annotate what type of list they want. After the user then annotates it as type int, I think it would be strange to return a float tensor. What warning would you suggest ?

Reply:
I think a general rule of thumb should be that code in script should run the same as code without script. If the user annotated the list as int without script, normal pytorch would still return an empty float tensor -- because of that discrepancy, we should warn when `torch.tensor([])` returns something that is not a float tensor (or the default tensor type)

Reply:
Added warning - let me know what you think

Reply:
Looks better now! One suggestion:

`torch.empty([])` can also return a `DoubleTensor` (as opposed to float tensor) if one changed the default dtype: https://pytorch.org/docs/master/torch.html#torch.set_default_dtype

Maybe you should warn that the "list will create a tensor of default floating point type (currently <>) in python but a tensor of type <> in torchscript"